vitessce/widget.py

```diff
@@ -36,6 +38,52 @@ def is_port_in_use(port):
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
         return s.connect_ex(('localhost', port)) == 0
 
+def get_port_and_routes(config, port, next_port, proxy=False, base_url=None):
+    if port is None:
+        use_port = next_port
+        next_port += 1
+        port_tries = 1
+        while is_port_in_use(use_port) and port_tries < MAX_PORT_TRIES:
+            use_port = next_port
+            next_port += 1
+            port_tries += 1
+    else:
+        use_port = port
+    
+    if base_url is None:
+        if proxy:
+            if importlib.util.find_spec('jupyter_server_proxy') is None:
+                raise ValueError("To use the widget through a proxy, jupyter-server-proxy must be installed.")
+            base_url = f"proxy/{use_port}"
+        else:
+            base_url = f"http://localhost:{use_port}"
+
+    config_dict = config.to_dict(base_url=base_url)
```

Comment:
Little concerned about returning the `config_dict` in this function given its name, but I do not see a better option that doesn't involve also returning the `base_url` (which is also not int he name of this function)